ProfileView/src/index.ts

```diff
@@ -157,120 +214,69 @@ export default class CjaasProfileWidget extends LitElement {
       method: "POST",
       headers: {
         "Content-type": "application/json",
-        Authorization: "SharedAccessSignature " + this.profileWriteToken
+        Authorization: "SharedAccessSignature " + this.profileToken
       },
       data
     };
     return axios(url, options)
       .then((x: AxiosResponse) => x.data)
       .then((_profile: ProfileFromSyncAPI) => {
-        this.profile = this.defaultTemplate.Attributes.map(
-          (attribute: any, i: number) => {
-            // if attribute is of tab type
-            // save journey events as well
-            let journeyEvents = null;
-            if (
-              attribute.type === "tab" ||
-              attribute.widgetAttributes?.type === "tab"
-            ) {
-              try {
-                journeyEvents = JSON.parse(
-                  _profile.attributeView[i].journeyEvents || "null"
-                );
-              } catch {
-                console.error("Error while parsing Journey Event");
-              }
-            }
-
-            return {
-              query: attribute,
-              result: _profile.attributeView[i].result.split(","),
-              journeyEvents
-            };
-          }
-        ) as Profile;
-
-        this.showSpinner = false;
+        this.profileData = this.parseResponse(this.defaultTemplate.attributes)
         this.requestUpdate();
       })
       .catch((err: Error) => {
         console.error("Could not load the Profile Data. ", err);
-        this.profile = undefined;
-        this.showSpinner = false;
+        this.profileData = undefined;
         this.requestUpdate();
       });
-  }
+    }
 
-  getProfileFromTemplateId() {
-    const url = `${this.baseURL}/v1/journey/views?viewId=${this.templateId}&personId=${this.customer}`;
 
-    this.showSpinner = true;
+  getProfileFromTemplateId() {
+    const url = `${this.baseURL}/v1/journey/views:build?templateId=${this.templateId}&personId=${this.customer}`;
 
     const options: AxiosRequestConfig = {
       url,
       method: "POST",
       headers: {
         "Content-type": "application/json",
-        Authorization: "SharedAccessSignature " + this.profileReadToken,
```

Comment:
Why are we renaming it? Old name informed us about the permission required as well.

Reply:
@venkateshwarv simplification, also. There's only one token passed to the widget, and it both reads and writes. I feel like it's overly complex to ak for one token to read and one to write, when you can get both in one token, therefore, your single "profile"  token

Reply:
@venkateshwarv  Actually, I don't know if it's worth fretting too much over right now . . . the next iteration is going to negate specific SAS tokens anyway since they will be generated with internal logic and more out of site from users